etl/steps/data/garden/oecd/2023-05-19/co2_air_transport.py

```diff
@@ -0,0 +1,307 @@
+    ds_garden = create_dataset(dest_dir, tables=[tb_garden], default_metadata=None)
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+    log.info("co2_air_transport.end")
+
+
+def ukraine_fill_war_for_reg_agg(df):
+    """
+    Fill missing values for 'Ukraine' in the DataFrame with zero values for the 'TER_INT_m'
+    column for concatenating regional emissions (then reset back to NaN) (2023 war so technically zero emissions and shouldn't be misleading for regional aggregates)
+
+    Args:
+        df (pd.DataFrame): Input DataFrame containing CO2 transport data.
+
+    Returns:
+        pd.DataFrame: DataFrame with missing values filled for 'Ukraine' in the 'TER_INT_m' column.
+
+    """
+    # Get a list of unique countries
+    unique_countries = df["country"].unique()
+
+    # Get the range of years that exist in the 'year' column
+    all_years = df["year"].unique()  # Inclusive range of years
+
+    # Create a DataFrame with all possible combinations of years and countries
+    new_index = pd.MultiIndex.from_product([all_years, unique_countries], names=["year", "country"])
+    new_df = pd.DataFrame(index=new_index).reset_index()
+
+    # Merge the new DataFrame with the original DataFrame
+    merged_df = pd.merge(new_df, df, on=["year", "country"], how="left")
+
+    # Optionally, sort the resulting DataFrame by the 'year' and 'country' columns
+    merged_df = merged_df.sort_values(["year", "country"]).reset_index(drop=True)
+
+    # Set values to NaN for missing years
+    merged_df.loc[merged_df["year"] < merged_df["year"].min(), "TER_INT_m"] = float("nan")
+    merged_df.loc[merged_df["year"] > merged_df["year"].max(), "TER_INT_m"] = float("nan")
+
+    # Create a Boolean mask to identify rows where the year is outside the desired range
+    year_mask = (merged_df["year"] < 2015) | (merged_df["year"] > 2023)
+
+    # Create a Boolean mask to identify rows where the country is 'Ukraine'
+    country_mask = merged_df["country"] == "Ukraine"
+
+    # Combine the masks using the logical AND operator
+    combined_mask = year_mask & country_mask
+
+    # Create the 'ter_int_m_filled_ukraine' column and fill it with the original 'TER_INT_m' values
+    merged_df["ter_int_m_filled_ukraine"] = merged_df["TER_INT_m"]
+
+    # Fill NaN values with 0s in the 'ter_int_m_filled_ukraine' column where the year is outside the range and the country is 'Ukraine'
+    merged_df.loc[combined_mask, "ter_int_m_filled_ukraine"] = merged_df.loc[
+        combined_mask, "ter_int_m_filled_ukraine"
+    ].fillna(0)
+
+    merged_df[["year", "ter_int_m_filled_ukraine"]][country_mask]
+
+    return merged_df
+
+
+def update_values(row):
+    """
+    Update values in the row based on the country.
+
+    Args:
+        row (pd.Series): A row of a DataFrame.
+
+    Returns:
+        pd.Series: Updated row.
+
+    """
+    regions_ = ["North America", "South America", "Europe", "Africa", "Asia", "Oceania"]
```

Comment:
(Not sure why these `regions_` are different from `regions` above. It would be nice to explain why `European Union (27)` is missing here.)

Reply:
actually I am not really using European Union anymore :) that was for smth I was trying out before but no longer need it so will just get rid off it :)